### PR TITLE
Added sklearn logging and checkpointing

### PR DESCRIPTION
**Pull request by DriessenA**
Adding sklearn logging and checkpointing. Additionally, some minor additions to work with the CAR data

### CODE REVIEW

File: cmonge/checkpointing/__init__.py
Comment (jannisborn) [inline cmonge/checkpointing/__init__.py:1]: Let's remove this empty folder

File: cmonge/trainers/ot_trainer.py [hunk 1]
```diff
@@ -94,20 +101,28 @@ def evaluate(
...
 
-            log_metrics(self.metrics, target, transport)
+            log_metrics(
+                self.metrics, target, transport, mlflow_logging=self.mlflow_logging
+            )
             if enum > n_samples:
                 break
 
-        log_mean_metrics(self.metrics)
+        log_mean_metrics(self.metrics, mlflow_logging=self.mlflow_logging)
```
Comment (jannisborn) [inline cmonge/trainers/ot_trainer.py:113]: The `mlflow_logging` attribute is not defined in this base class. Either define it in constructor (you can add it as a boolean with default False for full backwards compatability) or move this line out of the base class to the child classes

File: cmonge/trainers/ot_trainer.py [hunk 2]
```diff
@@ -130,32 +152,69 @@ def setup(
...
+            self.solver = map_estimator.MapEstimator(
+                num_genes,
+                fitting_loss=fitting_loss,
+                regularizer=regularizer,
+                model=model,
+                optimizer=optimizer,
+                regularizer_strength=1,
+                num_train_iters=num_train_iters,
+                logging=True,
+                valid_freq=100,  # mlflow logging not implemented here
```
Comment (jannisborn) [inline cmonge/trainers/ot_trainer.py]: I would add a `logger.warning("MLFlow not implemented for ..."` here

**Review by jannisborn — CHANGES_REQUESTED** [review]
great work overall, both in code style but also in ensuring backwards compatibility. Just some small remarks then we can merge

File: cmonge/checkpointing/__init__.py
Comment (DriessenA) [reply cmonge/checkpointing/__init__.py:1]: The folder isn't empty and the init is needed?

Comment (jannisborn) [reply cmonge/checkpointing/__init__.py:1]: Maybe you have unpushed changes but I dont see any code in this folder (just an empty init and another empty `.py`), hence my suggestion to delete

File: cmonge/trainers/ot_trainer.py [hunk 3]
```diff
@@ -13,19 +13,24 @@
...
+from orbax.checkpoint import CheckpointManagerOptions, CheckpointManager
+from orbax.checkpoint.args import StandardSave
 
 
 class AbstractTrainer:
     """Abstract class for neural OT traininig."""
 
-    def __init__(self, jobid: int, logger_path: Path) -> None:
+    def __init__(
+        self, jobid: int, logger_path: Path, mlflow_logging: bool = False
```
Comment (jannisborn) [inline cmonge/trainers/ot_trainer.py:32]: Now you're passing the variable but you're not registering as attribute in the constructor

File: cmonge/trainers/ot_trainer.py [hunk 4]
```diff
@@ -153,6 +155,7 @@ def setup(
         }
 
         # Save for use in `evaluate`
+        # Potentially overwrite AbstractTrainer attribute
```
Comment (jannisborn) [inline cmonge/trainers/ot_trainer.py]: This is not necessary, just move the attribute to the constructor of this class and then pass it down to the parent class via `super().__init__()`

**Review by jannisborn — APPROVED** [review]

**Pull request merged**